src/operator.ts

```diff
@@ -211,6 +211,51 @@ export default abstract class Operator {
         this._logger.info(`watching resource ${id}`);
     }
 
+    /**
+     * Watch a namespaced Kubernetes resource.
+     * @param group The group of the resource or an empty string for core resources
+     * @param version The version of the resource
+     * @param plural The plural name of the resource
+     * @param namespace The namespace which the resource is in
+     * @param onEvent The async callback for added, modified or deleted events on the resource
+     */
+    protected async watchNamespacedResource(group: string, version: string, plural: string, onEvent: (event: ResourceEvent) => Promise<void>, namespace?: string): Promise<void> {
```

Comment:
I would just modify the existing `watchResource()` to include the new optional `namespace?` parameter. It's backwards compatible after all.

Also make sure to update `README.md` with the new parameter.

Reply:
You're right, I will update the MR